### PR TITLE
Add perf benchmarking tests for recent LibG improvements

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose
https://jira.autodesk.com/browse/DYN-3123

Add perf tests for recent LibG improvements:
- Curve.ParameterAtPoint for PolyCurve
- Point.PruneDuplicates 

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### CODE REVIEW

**Comment by aparajit-pratap** [comment]
@mmisol could you share files for Point.PruneDuplicates and Surface.ByRuledLoft workflows.

File: tools/Performance/DynamoPerformanceTests/README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 | PythonJSONParsing       | 2.1 or later  | Strings, Dictionary, JSON, Python                                             | This graph uses JSON nodes, which were introduced in 2.1 | 
 | TSplinesSrf_to_Mesh     | 2.x           | Points, Meshes                                                                |                                                          | 
 | Vase                    | 2.x           | Curves, Surfaces, Solids, List Operations, Map/Combine                        |                                                          | 
+| CurveParameterAtPointQA_2.7.0_1 | 2.x   | Curve.ParameterAtPoint for PolyCurve                                          |                                                          |
```
Comment (mjkkirschner) [inline tools/Performance/DynamoPerformanceTests/README.md:23]: are more graphs going to be added?

**Comment by mjkkirschner** [comment]
I think you'll also need to update the base line references here:
https://git.autodesk.com/Dynamo/DynamoPerformance-CI/blob/master/baselines/DynamoPerformanceTests.DynamoModelPerformanceTestBase-report.csv

**Comment by aparajit-pratap** [comment]
> I think you'll also need to update the base line references here:
> https://git.autodesk.com/Dynamo/DynamoPerformance-CI/blob/master/baselines/DynamoPerformanceTests.DynamoModelPerformanceTestBase-report.csv

By running the benchmarks locally?

**Comment by mjkkirschner** [comment]
@aparajit-pratap no - that will give you a strange baseline based on your machine - once you merge this PR here, then you should get a run on the CI machine of your new tests which will output a new data file with mean, std deviation etc - then you can grab that file from jenkins and update the baseline with the latest data.

**Comment by aparajit-pratap** [comment]
> @aparajit-pratap no - that will give you a strange baseline based on your machine - once you merge this PR here, then you should get a run on the CI machine of your new tests which will output a new data file with mean, std deviation etc - then you can grab that file from jenkins and update the baseline with the latest data.

Ok, that's what I thought.

**Review by mjkkirschner — APPROVED** [review]

**Pull request merged**